src/rpc/budget.cpp

```diff
@@ -48,73 +48,85 @@ void budgetToJSON(CBudgetProposal* pbudgetProposal, UniValue& bObj)
     if (nPaymentCount < 1)
         throw std::runtime_error("Invalid payment count, must be more than zero.");
 
     // Start must be in the next budget cycle
     if (pindexPrev != NULL) nBlockMin = pindexPrev->nHeight - pindexPrev->nHeight % Params().GetBudgetCycleBlocks() + Params().GetBudgetCycleBlocks();
 
-    int nBlockStart = params[3].get_int();
+    nBlockStart = params[3].get_int();
     if (nBlockStart % Params().GetBudgetCycleBlocks() != 0) {
         int nNext = pindexPrev->nHeight - pindexPrev->nHeight % Params().GetBudgetCycleBlocks() + Params().GetBudgetCycleBlocks();
         throw std::runtime_error(strprintf("Invalid block start - must be a budget cycle block. Next valid block: %d", nNext));
     }
 
-    int nBlockEnd = nBlockStart + Params().GetBudgetCycleBlocks() * nPaymentCount; // End must be AFTER current cycle
+    int nBlockEnd = nBlockStart + (Params().GetBudgetCycleBlocks() * nPaymentCount); // End must be AFTER current cycle
 
     if (nBlockStart < nBlockMin)
         throw std::runtime_error("Invalid block start, must be more than current height.");
 
     if (nBlockEnd < pindexPrev->nHeight)
         throw std::runtime_error("Invalid ending block, starting block + (payment_cycle*payments) must be more than current height.");
 
-    CBitcoinAddress address(params[4].get_str());
+    address = params[4].get_str();
     if (!address.IsValid())
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid PIVX address");
 
+    nAmount = AmountFromValue(params[5]);
+}
+
+UniValue preparebudget(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 6)
+        throw std::runtime_error(
+            "preparebudget \"proposal-name\" \"url\" payment-count block-start \"pivx-address\" monthy-payment\n"
+            "\nPrepare proposal for network by signing and creating tx\n"
+
+            "\nArguments:\n"
+            "1. \"proposal-name\":  (string, required) Desired proposal name (20 character limit)\n"
+            "2. \"url\":            (string, required) URL of proposal details (64 character limit)\n"
+            "3. payment-count:    (numeric, required) Total number of monthly payments\n"
+            "4. block-start:      (numeric, required) Starting super block height\n"
+            "5. \"pivx-address\":   (string, required) PIVX address to send payments to\n"
+            "6. monthly-payment:  (numeric, required) Monthly payment amount\n"
+
+            "\nResult:\n"
+            "\"xxxx\"       (string) proposal fee hash (if successful) or error message (if failed)\n"
+
+            "\nExamples:\n" +
+            HelpExampleCli("preparebudget", "\"test-proposal\" \"https://forum.pivx.org/t/test-proposal\" 2 820800 \"D9oc6C3dttUbv8zd7zGNq1qKBGf4ZQ1XEE\" 500") +
+            HelpExampleRpc("preparebudget", "\"test-proposal\" \"https://forum.pivx.org/t/test-proposal\" 2 820800 \"D9oc6C3dttUbv8zd7zGNq1qKBGf4ZQ1XEE\" 500"));
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
```

Comment:
Probably best to rebase this off of `CaveSpectre11:RPCBudgetCrash` branch (#965) to have a smoother merge and not forget, for example, the pwalletMain check here:
```cpp
if (!pwalletMain) {
    throw std::runtime_error("Try again after wallet is fully started");
}

Reply:
oh they definitely won't forget; I wasn't certain that was procedure... I can rebase off an existing open PR and have it as a dependency of that merge?  (definitely makes my life easier)

Reply:
Yes, exactly. Much easier that way :+1:

Reply:
ok sweet.  as soon as I finish this build of 950; I will rebase both.